pallets/loans/src/functions.rs

```diff
@@ -224,27 +213,24 @@ impl<T: Config> Pallet<T> {
 		// ensure owner is the loan owner
 		Self::check_loan_owner(pool_id, loan_id, owner.clone())?;
 
-		LoanInfo::<T>::try_mutate(
+		Loan::<T>::try_mutate(
 			pool_id,
 			loan_id,
-			|maybe_loan_info| -> Result<bool, DispatchError> {
-				let mut loan_info = maybe_loan_info.take().ok_or(Error::<T>::MissingLoan)?;
+			|maybe_loan| -> Result<bool, DispatchError> {
+				let mut loan = maybe_loan.take().ok_or(Error::<T>::MissingLoan)?;
```

Comment:
I would love to get rid of all the `let mut loan = maybe_loan.take().ok_or(Error::<T>::MissingLoan)?;` and rather work on the references instead via `let mut loan = maybe_loan.as_ref().ok_or(Error::<T>::MissingLoan)?;`.

Replacing the given Option with a None feels bad.

Reply:
I agree, I much prefer the `as_mut()` pattern we use in the `Pools` pallet

Reply:
Just changed this @mustermeiszer @branan